libs/fluid-elements/virtual-scroll-container/src/lib/virtual-scroll-state.ts

```diff
@@ -0,0 +1,291 @@
+/**
+ * @license
+ * Copyright 2020 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface FluidVirtualScrollOptions {
+  itemsBatchSize?: number;
+  itemsBufferSize?: number;
+}
+
+export interface FluidVirtualScrollRenderedItemRange {
+  first: number;
+  last: number;
+}
+
+// Number of items to render each time until the container is filled up
+const VIRTUAL_SCROLL_BATCH_SIZE = 5;
+// Number of items to render in addition to the initially rendered items
+const VIRTUAL_SCROLL_BUFFER_SIZE = 3;
+
+export class VirtualScrollState {
+  /** Average item height */
+  avgItemHeight: number;
+
+  /**
+   * Index of the first and last rendered item.
+   *
+   * Important note: Never attempt to enforce to alter the object returned by
+   * this getter nor rely on its state after {@link #updateStartScrollBuffer()}
+   * or {@link #updateEndScrollBuffer()} have been called again!
+   * In case you need to store that information past this point, always create
+   * a copy right away instead.
+   * @ ChMa: pls explain again why it should be copied? I forgot what the potential issue was...
+   */
+  renderedItemsRange: FluidVirtualScrollRenderedItemRange = Object.seal({
+    first: -1,
+    last: -1,
+  });
+
+  /** Empty placeholder before the rendered items to correctly render the scrollbar */
+  startPlaceholderSize = 0;
+
+  /** Empty placeholder after the rendered items to correctly render the scrollbar */
+  endPlaceholderSize = 0;
+
+  /** Number of currently rendered start buffer items */
+  startItemsBufferSize = 0;
+
+  /** Number of currently rendered end buffer items */
+  endItemsBufferSize = 0;
+
+  /** Index of the item to observe for adjusting the start buffer */
+  bufferStartObservedItemIndex = -1;
+
+  /** Index of the item to observe for adjusting the end buffer */
+  bufferEndObservedItemIndex = -1;
+
+  /** Total number of items in the virtual scroll panel */
+  set itemsCount(value: number) {
```

Comment:
Would recommend always having a getter/setter (adding the getter here) pair, unless it is a special case that requires it.